arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "path": {
+          "description": "Gets or sets the path.",
+          "type": "string"
+        },
+        "endpoint": {
+          "description": "Gets or sets the endpoint.",
+          "type": "string"
+        },
+        "internalAuthorizationPolicies": {
+          "description": "Gets or sets the internal authorization rules.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharedAccessAuthorizationRule"
+          }
+        },
+        "authorizationPolicies": {
+          "description": "Gets or sets the authorization rules.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharedAccessAuthorizationRule"
+          }
+        }
+      }
+    },
+    "StorageEndpointProperties": {
+      "description": "Contains the user facing view of the properties of the Storage Endpoint provided by the customer for file upload",
+      "type": "object",
+      "properties": {
+        "sasTtlAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "connectionString": {
+          "description": "Account key credentials for storage account selected by customer for uploading files",
+          "type": "string"
+        },
+        "containerName": {
+          "description": "Root container name where all files will be uploaded",
+          "type": "string"
+        }
+      }
+    },
+    "MessagingEndpointProperties": {
+      "description": "Contains the user facing view of the properties of the Messaging Endpoints used by this IoT Hub",
+      "type": "object",
+      "properties": {
+        "lockDurationAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "ttlAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "maxDeliveryCount": {
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "CloudToDeviceProperties": {
+      "description": "Contains user facing view of Iot Hub Cloud-To-Device messaging properties",
+      "type": "object",
+      "properties": {
+        "maxDeliveryCount": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "defaultTtlAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "feedback": {
+          "$ref": "#/definitions/FeedbackProperties",
+          "description": "Gets the feedback properties."
+        }
+      }
+    },
+    "OperationsMonitoringProperties": {
+      "description": "Contains the user facing view of the Operation Monitoring properties",
+      "type": "object",
+      "properties": {
+        "events": {
+          "type": "object",
+          "additionalProperties": {
+            "enum": ["None", "Error", "Information"],
```

Comment:
@rkmanda - Oh no please don't. We are commenting in good spirit to make the swagger doc better, so that the generated client and the documentation is much better. 
It is my personal opinion that devs from the service team should manually create the swagger spec (prototype) only for the first time (even before writing the server). This gives them some time to think how does their API actually look. The swagger spec eventually becomes the contract between the server and the client.

Successive iterations can be auto generated using swashbuckle or any other plugin. The spec generated so far is good and it would be great if you can modify the plugin to incorporate all our comments. We would like to make this a model for other service teams. They can use the stuff that you did so that they don't have to reinvent the wheel.